toolchain/check/testdata/interop/cpp/no_prelude/function_decl_inline.carbon

```diff
@@ -0,0 +1,125 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/interop/cpp/no_prelude/function_decl_inline.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/interop/cpp/no_prelude/function_decl_inline.carbon
+
+// ============================================================================
+// inline_function_decl_with_definition
+// ============================================================================
+
+// --- inline_function_decl_with_definition.h
+
+inline void foo() {}
+
+// --- import_inline_function_decl_with_definition.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "inline_function_decl_with_definition.h";
+
+fn MyF() {
+  Cpp.foo();
+}
+
+// ============================================================================
+// inline_function_decl_without_definition
+// ============================================================================
+
+// --- inline_function_decl_without_definition.h
+
+inline void foo();
+
+// --- todo_import_inline_function_decl_without_definition.carbon
```

Comment:
```suggestion
// --- todo_fail_import_inline_function_decl_without_definition.carbon
```

Reply:
Why should it fail?
IIUC, warnings should not trigger a failure.

Reply:
1. Warnings cause failures.
2. It should probably be an error in Carbon, not a warning. We want Carbon to be more strict about correctness than C++.

Reply:
1. Right, but in Carbon tests I've noticed warnings don't make tests have `fail_` prefix.
2. If that's an imported C++ code, I'm not sure we want to require the user to change the C++ code to make Carbon happy, but up to you. I can clarify this needs to be an error. However, note that if we end up making the C++ compilation trigger this, it will be a C++ warning, and we typically convert them to Carbon warnings, though we can probably decide to do it on a per warning case if it's worth the special casing.

Reply:
> 2. If that's an imported C++ code, I'm not sure we want to require the user to change the C++ code to make Carbon happy, but up to you.

I'm trying to understand here, why do you think this is to "make Carbon happy" versus diagnosing invalid code? Let's discuss inline functions in the next meeting.

> Changed to `todo_fail_` and clarified we want an error.

Did you forget to push changes?

Reply:
Yes, forgot to push the changes, sorry, done now.

Regarding the diagnosis, I think being consistent with C++ when it comes to interop with C++ makes a lot of sense and would make migration easier (people could have flags or tools that rely on C++ behavior).
Being more strict has significant benefits, but we need to consider the tradeoffs with migration difficulty.
I don't feel strongly about this and actually not sure what the guideline Carbon takes on this tradeoff, just sharing my opinion.